pandas/core/dtypes/concat.py

```diff
@@ -152,11 +152,11 @@ def is_nonempty(x) -> bool:
             target_dtype = find_common_type([x.dtype for x in to_concat])
             to_concat = [_cast_to_common_type(arr, target_dtype) for arr in to_concat]
 
-        if isinstance(to_concat[0], ExtensionArray) and axis == 0:
+        if isinstance(to_concat[0], ExtensionArray):  # and axis == 0:
```

Comment:
Related to https://github.com/pandas-dev/pandas/pull/35038#discussion_r483613738, testing if this is actually needed (need to clean this up, of course)